### DISCUSSION TITLE
How do you create two mutually exclusive yet required options in CLI11?

### DISCUSSION THREAD

**Discussion by Coderx7** [CLIUtils/CLI11#583] [Q&A]
This is a follow-up question from [here](https://github.com/CLIUtils/CLI11/issues/88). It seems this is not [clearly?] discussed in the documentations. so here it is again,  How can we have two required mutually exclusive options in CLI11?

**Comment by phlptp** [comment]
Just in case this has not been answered yet
the way to do this is that the options are placed in an option group.  

```cpp
group=app.add_option_group("subgroup");
group->add_flag("--cpu", isOnCpu, "Run the computation on the CPU")->ignore_case();
    group->add_flag("--gpu", isOnGpu, "Run the computation on the GPU")->ignore_case();

group->require_option(1); 
```

The require_option(1)  will state that 1 and only 1 option from the group is required.  Any more or less is an error.
